File: frontend/src/constants/appeal.ts
export const APPEAL_STATUSES = [
  { value: 'pending', label: '待审核', type: 'warning' },
  { value: 'approved', label: '已通过', type: 'success' },
  { value: 'rejected', label: '已驳回', type: 'danger' },
] as const

export const APPEAL_ACTIONS = [
  { value: 'approve', label: '审核通过（撤销信誉分变化）' },
  { value: 'reject', label: '驳回（信誉分不变）' },
] as const

export function appealStatusLabel(value: string): string {
  return APPEAL_STATUSES.find((s) => s.value === value)?.label ?? value
}

export function appealStatusType(value: string): string {
  return APPEAL_STATUSES.find((s) => s.value === value)?.type ?? 'info'
}

File: backend/internal/constants/appeal.go
package constants

// AppealStatus defines credit-appeal state machine values shared with the frontend.
const (
	AppealStatusPending  = "pending"
	AppealStatusApproved = "approved"
	AppealStatusRejected = "rejected"
)

// AppealStatuses lists all valid appeal statuses in flow order.
var AppealStatuses = []string{
	AppealStatusPending, AppealStatusApproved, AppealStatusRejected,
}

// IsAppealStatus reports whether the given status is valid.
func IsAppealStatus(s string) bool {
	for _, v := range AppealStatuses {
		if v == s {
			return true
		}
	}
	return false
}

// AppealStatusText returns the Chinese label of an appeal status.
func AppealStatusText(s string) string {
	switch s {
	case AppealStatusPending:
		return "待审核"
	case AppealStatusApproved:
		return "已通过"
	case AppealStatusRejected:
		return "已驳回"
	default:
		return "未知"
	}
}

// AppealAction defines admin review actions shared with the frontend.
const (
	AppealActionApprove = "approve"
	AppealActionReject  = "reject"
)

// AppealActions lists all valid admin actions.
var AppealActions = []string{AppealActionApprove, AppealActionReject}

// IsAppealAction reports whether the given review action is valid.
func IsAppealAction(a string) bool {
	for _, v := range AppealActions {
		if v == a {
			return true
		}
	}
	return false
}
